### PR TITLE
TEP-0114: Remove support for v1alpha1.Run

### PR DESCRIPTION
**Pull request by lbernick**
v1alpha1.Run has been replaced by v1beta1.CustomRun, and the PipelineRun controller no longer creates v1alpha1.Run objects. This commit removes support for v1alpha1.Run in the PipelineRun controller. It maintains the API fields for v1alpha1.Run for use by clients.

Note: Retries behavior has changed between v1alpha1.Run and v1beta1.CustomRun. The PipelineRun controller does not consider whether CustomRuns have retries remaining to determine whether they have completed; it only considers their status.

/kind cleanup
Closes https://github.com/tektoncd/pipeline/issues/6498
Closes https://github.com/tektoncd/pipeline/issues/5870
Closes https://github.com/tektoncd/pipeline/issues/5863

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
action required: v1alpha1.Run objects are no longer supported. You must upgrade to v1beta1.CustomRun before upgrading to this release. See https://github.com/tektoncd/pipeline/blob/main/docs/migrating-v1alpha1.Run-to-v1beta1.CustomRun.md for migration instructions
```

### CODE REVIEW

**Comment by Yongxuanzhang** [comment]
Thanks! I wonder do need to remove `run` in `config/200-clusterrole.yaml`?

**Comment by lbernick** [comment]
> Thanks! I wonder do need to remove `run` in `config/200-clusterrole.yaml`?

nice catch, thanks!

File: pkg/reconciler/pipelinerun/pipelinerun_test.go [hunk 1]
```diff
@@ -1046,17 +1045,17 @@ func TestReconcileOnCompletedPipelineRun(t *testing.T) {
 	// in the PipelineRun status, that the completion status is not altered, that not error is returned and
 	// a successful event is triggered
 	taskRunName := "test-pipeline-run-completed-hello-world-task-run"
-	runName := "test-pipeline-run-completed-hello-world-run"
+	customRunName := "test-pipeline-run-completed-hello-world-run"
```
Comment (jerop) [inline pkg/reconciler/pipelinerun/pipelinerun_test.go:1048]:
```suggestion
	customRunName := "test-pipeline-run-completed-hello-world-custom-run"
```

**Pull request merged**